cognite_toolkit/_cdf_tk/templates/data_classes/_config_yaml.py

```diff
@@ -48,7 +48,9 @@ def load(cls, data: dict[str, Any], build_name: str) -> Environment:
                     if MODULE_PATH_SEP in selected
                     else selected
                     for selected in data["selected_modules_and_packages"]
-                ],
+                ]
+                if data["selected_modules_and_packages"]
+                else [],
```

Comment:
Not sure what this achieves, it looks like the same as was already there. Do you try to make it optional to provide |selected_modules_and_packages`?

Reply:
What about just changing the line above to `data.get("selected_modules_and_packages", [])`?

Reply:
This code covers the rare case when you have the key "selected_modules_and_packages" but no items beneath it, I don't think yours does?

Reply:
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/pal.ronning@cognitedata.com/Local/cdf-project-templates/cognite_toolkit/_cdf_tk/templates/data_classes/_config_yaml.py", line 133, in load
    environment = Environment.load(data["environment"], build_env_name)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/pal.ronning@cognitedata.com/Local/cdf-project-templates/cognite_toolkit/_cdf_tk/templates/data_classes/_config_yaml.py", line 50, in load
    for selected in data["selected_modules_and_packages", []]
                    ~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: unhashable type: 'list'

Reply:
Ah now I get it. I assumed `pyyaml` parsed an empty list, but of course it cannot now whether it is a list or a mapping, so it parses None.

I though you had this
```python
data = {"selected_modules_and_packages": []}
```
while you actually got this:
```python
data = {"selected_modules_and_packages": None}
```

Reply:
Anyways if you want to 🏌

then , this also works: 
```python
for selected in data["selected_modules_and_packages"] or []
```

Reply:
🏌️ ⛳